pkg/prometheus/statefulset.go

```diff
@@ -658,7 +658,7 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMapName
 
 		thanosArgs := []string{
 			"sidecar",
-			fmt.Sprintf("--prometheus.url=http://%s:9090", c.LocalHost),
+			fmt.Sprintf("--prometheus.url=http://%s:9090%s", c.LocalHost, webRoutePrefix),
```

Comment:
I suggest to handle `webRoutePrefix` similar to other parts in this function, where we use `path.Clean`.
```suggestion
			fmt.Sprintf("--prometheus.url=http://%s:9090%s", c.LocalHost, path.Clean(webRoutePrefix)),
```

https://github.com/coreos/prometheus-operator/blob/master/pkg/prometheus/statefulset.go#L429